python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -474,6 +474,25 @@ def get_map_transformer(self) -> MapTransformer:
         return self._map_transformer
 
     def shutdown(self, force: bool = False):
+        if force:
+            tasks: List[OpTask] = list(self._data_tasks.values()) + list(
+                self._metadata_tasks.values()
+            )
+
+            # Interrupt all (still) running tasks immediately
+            for task in tasks:
+                ray.cancel(task.get_waitable(), force=True)
+
+            # Wait for all tasks to get cancelled before returning
+            for task in tasks:
+                try:
+                    ray.get(task.get_waitable())
+                except ray.exceptions.RayError:
+                    # Cancellation either succeeded, or the task might have already
+                    # failed with a different error, or cancellation failed.
+                    # In all cases, we swallow the exception.
+                    pass
```

Comment:
IIRC, actor tasks are not cancellable. (I don't remember if cancelling them will be a no-op or will raise errors).
That's why the task cancellation is put in the task-pool subclass.

Reply:
Yeah, it will be no-op for executing actor tasks but it will cancel pending ones

https://github.com/ray-project/ray/blob/0a0aad6b00dabc25134d75d7cbdc34ebc3ecd8c1/python/ray/_private/worker.py#L3159